File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/DefaultKinveyDeleteCallback.java
package com.kinvey.androidTest.callback;

import com.kinvey.android.callback.KinveyDeleteCallback;

import java.util.concurrent.CountDownLatch;

/**
 * Created by yuliya on 10/10/17.
 */

public class DefaultKinveyDeleteCallback implements KinveyDeleteCallback{

    private CountDownLatch latch;
    private Integer result;
    private Throwable error;

    public DefaultKinveyDeleteCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(Integer result) {
        this.result = result;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    void finish() {
        latch.countDown();
    }

    public Integer getResult() {
        return result;
    }

    public Throwable getError() {
        return error;
    }


}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/DefaultKinveyDeleteCallback.kt
package com.kinvey.androidTest.callback

import com.kinvey.android.callback.KinveyDeleteCallback
import java.util.concurrent.CountDownLatch

/**
 * Created by yuliya on 10/10/17.
 */

class DefaultKinveyDeleteCallback(private val latch: CountDownLatch) : KinveyDeleteCallback {
    var result: Int? = null
        private set
    var error: Throwable? = null
        private set

    override fun onSuccess(result: Int?) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable?) {
        this.error = error
        finish()
    }

    internal fun finish() {
        latch.countDown()
    }

}